packages/host/tests/unit/ai-function-generation-test.ts

```diff
@@ -0,0 +1,318 @@
+  let adapter: TestRealmAdapter;
+  let realm: Realm;
+  let owner: any;
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    loader = (this.owner.lookup('service:loader-service') as LoaderService)
+      .loader;
+  });
+
+  setupLocalIndexing(hooks);
+  setupOnSave(hooks);
+  setupCardLogs(
+    hooks,
+    async () => await loader.import(`${baseRealm.url}card-api`),
+  );
+  setupServerSentEvents(hooks);
+
+  async function createCard(code: string) {
+    cardApi = await loader.import(`${baseRealm.url}card-api`);
+    adapter = new TestRealmAdapter({
+      'testcard.gts': code,
+    });
+    realm = await TestRealm.createWithAdapter(adapter, loader, owner, {
+      overridingHandlers: [
+        async (req: Request) => {
+          return sourceFetchRedirectHandle(req, adapter, testRealmURL);
+        },
+        async (req: Request) => {
+          return sourceFetchReturnUrlHandle(req, realm.maybeHandle.bind(realm));
+        },
+      ],
+    });
+    await realm.ready;
+    return await loadCard(
+      { name: 'TestCard', module: 'testcard' },
+      { loader: loader, relativeTo: new URL(testRealmURL) },
+    );
+  }
+
+  hooks.beforeEach(async function () {
+    owner = this.owner;
+  });
+
+  test(`generates a simple compliant schema for basic types`, async function (assert) {
+    let basicCard = await createCard(`
+    import StringField from 'https://cardstack.com/base/string';
+    import NumberField from 'https://cardstack.com/base/number';
+    import {
+        Component,
+        CardDef,
+        field,
+        contains,
+    } from 'https://cardstack.com/base/card-api';
+
+    export class TestCard extends CardDef {
+        static displayName = 'TestCard';
+        @field stringField = contains(StringField);
+        @field numberField = contains(NumberField);
+        @field computedField = contains(StringField, {
+        computeVia: function (this: TestCard) {
+            return 'generated';
+        },
+        });
+    }
+    `);
+    let schema = cardApi.generatePatchCallSpecification(basicCard);
+    assert.deepEqual(schema, {
+      type: 'object',
+      properties: {
+        thumbnailURL: { type: 'string' },
+        title: { type: 'string' },
+        description: { type: 'string' },
+        stringField: { type: 'string' },
+        numberField: { type: 'number' },
+      },
+    });
+  });
+
+  test(`generates a simple compliant schema when there are thunks`, async function (assert) {
+    let basicCard = await createCard(`
+    import StringField from 'https://cardstack.com/base/string';
+    import NumberField from 'https://cardstack.com/base/number';
+    import {
+        Component,
+        CardDef,
+        field,
+        contains,
+    } from 'https://cardstack.com/base/card-api';
+
+    export class TestCard extends CardDef {
+        static displayName = 'TestCard';
+        @field stringField = contains(() => StringField);
```

Comment:
no need to use the thunk `() => StringField`--that is only for linked fields. in fact this is a type error--but you cant see that since you are using strings to create cards.

Reply:
I don't have any special handling of this in the code so can remove the test as it was there to make sure things worked when people used thunks, but as a note this works and does so without an error if written in the code editor. I'm not sure if we can deal with that without glint support.

That also lets you build a recursive schema which works just fine until you try and then load anything.